ragengine/embedding/huggingface_local.py

```diff
@@ -0,0 +1,19 @@
+from llama_index.embeddings.huggingface import HuggingFaceEmbedding
+
+from .base import BaseEmbeddingModel
+
+
+class LocalHuggingFaceEmbedding(BaseEmbeddingModel):
+    def __init__(self, model_name: str):
+        self.model = HuggingFaceEmbedding(model_name=model_name) # TODO: Ensure/test loads on GPU (when available)
+
+    def get_text_embedding(self, text: str):
+        """Returns the text embedding for a given input string."""
+        return self.model.get_text_embedding(text)
+
+    def get_embedding_dimension(self) -> int:
+        """Infers the embedding dimension by making a local call to get the embedding of a dummy text."""
+        dummy_input = "This is a dummy sentence."
```

Comment:
I know this is for test functionality, but can we add the specific idea here? like "Test model functionality and output dimensions."?

Reply:
Actually, this method isn't for testing, but to initialize objects with the correct embedding dimension, which is why I think the current comment works fine